darts/models/forecasting/tft_model.py

```diff
@@ -406,18 +411,30 @@ def get_attention_mask_full(
 
     @staticmethod
     def get_attention_mask_future(
-        encoder_length: int, decoder_length: int, batch_size: int, device: str
+        encoder_length: int,
+        decoder_length: int,
+        batch_size: int,
+        device: str,
+        full_attention: bool,
     ) -> torch.Tensor:
         """
         Returns causal mask to apply for self-attention layer that acts on future input only.
+        The model will attend to all `False` values.
         """
-        # indices to which is attended
-        attend_step = torch.arange(decoder_length, device=device)
-        # indices for which is predicted
-        predict_step = torch.arange(0, decoder_length, device=device)[:, None]
-        # do not attend to steps to self or after prediction
-        decoder_mask = attend_step >= predict_step
-        # do not attend to past input
+        if full_attention:
+            # attend to entire past and future input
+            decoder_mask = torch.zeros(
+                (decoder_length, decoder_length), dtype=torch.bool, device=device
+            )
+        else:
+            # attend only to past steps relative to forecasting step in the future
+            # indices to which is attended
+            attend_step = torch.arange(decoder_length, device=device)
+            # indices for which is predicted
+            predict_step = torch.arange(0, decoder_length, device=device)[:, None]
+            # do not attend to steps to self or after prediction
```

Comment:
self is basically the self attention at the current prediction step. It will also ignore the prediction steps of next horizons (ahead of it's own predict step, self).

So we actually do not attend to steps **related** to the individual prediction and any future predictions.